src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -45,8 +45,9 @@ internal class UnixPwmChannel : PwmChannel
             _dutyCycleWriter = new StreamWriter(new FileStream($"{_channelPath}/duty_cycle", FileMode.Open, FileAccess.ReadWrite));
             _frequencyWriter = new StreamWriter(new FileStream($"{_channelPath}/period", FileMode.Open, FileAccess.ReadWrite));
 
-            SetFrequency(frequency);
-            DutyCycle = dutyCycle;
+            // If duty cycle is set to value 
+            _dutyCycle = dutyCycle;
+            SetFrequency(frequency, forceUpdateDutyCycle: true);
```

Comment:
Did you test this? IIRC by default in the unix driver frequency is set to 0 initially, so it will fail to set duty cycle first as it has to be a number which is lower than the frequency.

Reply:
Yes, I haven't had any issues locally yet. Also I don't set the frequency here, just changing the field - the setting of the duty cycle will actually happen after setting frequency

Reply:
Ok, seems this is actually a problem just after boot but not after you set it the first time. Logic without any issues was a bit tricky though but not seeing any problems on updated version